src/target/linux.rs

```diff
@@ -144,13 +200,63 @@ impl<'a> ReadMemory<'a> {
             )
         };
 
-        if bytes_read == -1 {
-            // fixme: return a proper error type
-            return Err(Box::new(nix::Error::last()));
+
+        if bytes_read < read_len as isize {
+            let maps = Process::new(self.pid.as_raw())?.maps()?;
```

Comment:
FYI #59 will add a `memory_maps` method. The returned `MemoryMap` type doesn't have a permission field, but it is easy to add.